diff --git a/src/qcommon/q_platform.h b/src/qcommon/q_platform.h
index 2c59e2777..3e61a5f07 100644
--- a/src/qcommon/q_platform.h
+++ b/src/qcommon/q_platform.h
@@ -218,6 +218,10 @@
 #define ARCH_STRING "sparc"
 #elif defined __arm__
 #define ARCH_STRING "arm" // __ARM_ARCH_'V'__ FIXME: add ARM version to the ARCH_STRING
+#elif defined __aarch64__
+#undef idx64
+#define idx64 1
+#define ARCH_STRING "aarch64"
 #elif defined __cris__
 #define ARCH_STRING "cris"
 #elif defined __hppa__
